dll/cpl/sysdm/environment.c

```diff
@@ -392,10 +824,20 @@ OnEditVariable(HWND hwndDlg,
                INT iDlgItem)
 {
     HWND hwndListView;
-    PVARIABLE_DATA VarData;
+    PDIALOG_DATA DlgData;
     LV_ITEM lvi;
     INT iItem;
+    INT DlgID = IDD_EDIT_VARIABLE;
+    INT iRet;
 
+    DlgData = GlobalAlloc(GPTR, sizeof(DIALOG_DATA));
+    
+    DlgData->bIsFriendlyUI = FALSE;
+    DlgData->dwSelectedValueIndex = -1;
+    DlgData->dwTextEditedValueIndex = -1;
+
+    DlgData->VarData = GlobalAlloc(GPTR, sizeof(VARIABLE_DATA));
```

Comment:
Why are you allocating new VarData here?
You are editing a variable, not creating a new one...